### PR TITLE
chore(gatsby-source-s3): AWS SDK for JavaScript v2 Going Into Maintenance Mode (#386)

### LINKED ISSUE #386: chore(gatsby-source-s3): AWS SDK for JavaScript v2 Going Into Maintenance Mode
## Describe the bug

When starting up a project with `gatsby-source-s3` installed, an error message is displayed. The error message is as follows:

> ERROR 
>
>(node:22436) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
>
>Please migrate your code to use AWS SDK for JavaScript (v3).
>For more information, check the migration guide at https://a.co/7PzMCcy
>(Use `node --trace-warnings ...` to show where the warning was created)

## To Reproduce

With a Gatsby v4 project:
1. `npm i gatsby-source-s3`
2. Run either `gatsby develop` or `gatsby build`.

## Expected behavior

Error messages should not be displayed when building the project.

### PR DESCRIPTION
**Pull request by h93xV2**
Question for maintainers, was it correct to just increment the major version in the package file or do I need to do something else to make sure the version is incremented?

## Description

Updated the AWS SDK dependency from v2 to v3 since projects using the dependency print an error message to the console when they start up. The error message explains that v2 will be going into maintenance mode in 2023. The update causes the error message to no longer be displayed. The update necessitated modifications to the integration tests as well.

The major version for the plugin was incremented since the object used to configure the AWS client has a slightly different shape in v3 compared to v2.

### Documentation

The README.md file was updated to show the new shape of the AWS config object. The main change is that credentials are under a named field.

## Related Issues

Fixes #386

### CODE REVIEW

File: integration-tests/source-s3/gatsby-config.js [hunk 1]
```diff
@@ -9,12 +9,14 @@ module.exports = {
...
-          endpoint: "localhost:9001",
+          credentials: {
+            accessKeyId: "minioadmin",
+            secretAccessKey: "minioadmin",
+          },
+          forcePathStyle: true,
+          endpoint: "http://localhost:9001",
           sslEnabled: false,
-          s3ForcePathStyle: true,
+          region: "us-east-1",
```
Comment (moonmeister) [inline integration-tests/source-s3/gatsby-config.js:19]: Why was this added?

File: packages/gatsby-source-s3/package.json [hunk 2]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "gatsby-source-s3",
   "description": "A Gatsby plugin to source objects and images from AWS S3",
-  "version": "3.2.4",
+  "version": "4.0.0",
```
Comment (moonmeister) [inline packages/gatsby-source-s3/package.json:4]: This will be handled by CI please revert the version change and run 'yarn cs' from the repo root to add a changelog and select major version release.

**Review by moonmeister — CHANGES_REQUESTED** [review]
Looking good so far, couple small things. Thanks for the PR, I'll kick off CI and see what the tests say!

**Comment by moonmeister** [comment]
You're also welcome to run 'yarn ac' from the root to add yourself as a code and docs contributor.

File: packages/gatsby-source-s3/package.json (hunk 2 above)
Comment (h93xV2) [reply packages/gatsby-source-s3/package.json:4]: Is there a pattern for additional commit messages? I kind of guessed based on what I saw in the project history for the first message.

Comment (moonmeister) [reply packages/gatsby-source-s3/package.json:4]: Commits, won't really, I use semantic commits but not required. For the change log please clearly state what changed and specifically what was a breaking change and point to the updated docs for config. The changeset file is markdown so you can always do more formatting and content after creating the file with the cli command.

File: integration-tests/source-s3/gatsby-config.js (hunk 1 above)
Comment (h93xV2) [reply integration-tests/source-s3/gatsby-config.js:19]: From what I understand, v3 doesn't support a default region for custom endpoints. There's this comment which references a feature request to add the functionality from a few years ago https://github.com/aws/aws-sdk-js-v3/issues/1845#issuecomment-754832210, but that feature hasn't been added yet.

The region property comes from the `RegionInputConfig` interface which denotes the property as optional with no default. Since the integration test is using a custom endpoint (localhost), the region property has to be in the config otherwise an error is thrown.

File: packages/gatsby-source-s3/package.json (hunk 2 above)
Comment (h93xV2) [reply packages/gatsby-source-s3/package.json]: I've reverted the version change and added a changelog with `yarn cs`.

**Review by moonmeister — APPROVED** [review]
Thanks, nice work. Assuming CI passes this will get merged.
